packages/headless/src/features/query-suggest/query-suggest-slice.ts

```diff
@@ -90,8 +92,27 @@ export const querySuggestReducer = createReducer(
 
         querySuggest.q = expression;
       })
+      .addCase(restoreSearchParameters, (state, action) => {
+        updateAllQuerySuggestSetQuery(state, action.payload.q ?? '');
```

Comment:
So it appears the querySuggestSetSlice query and the querySetSlice query are kept in sync, should we deprecate the querySuggestSetSlice query and use the querySetSlice in `buildQuerySuggestRequest`, this would imply adding `QuerySetSection` to `StateNeededByQuerySuggest`, is that breaking?

Reply:
Having one source of truth sounds good. I think it will be ok 👍

Reply:
I don't think I'll do it in the end since i'm seeing partial queries relying on it inside the same slice.